### PR TITLE
WIP: Blockchain-api: deals invoke

### PR DESCRIPTION
**Pull request by Sokel**
made handles to work with our contracts

### CODE REVIEW

File: accounts/identity.go [hunk 1]
```diff
@@ -160,9 +159,5 @@ func parseKeystoreUrl(path string) (string, error) {
 // return default keystore directory stored in in `.sonm` directory
 // if any error occurred .sonm directory will be in working dir
 func GetDefaultKeystoreDir() string {
-	rootDir, err := util.GetUserHomeDir()
-	if err != nil {
-		rootDir = ""
-	}
-	return filepath.Join(rootDir, ".sonm", "keystore")
+	return filepath.Join("etc", "sonm", "keystore")
```
Comment (sshaman1101) [inline accounts/identity.go]: I think it not a pretty good idea. Is there any reason to use `CWD/etc/sonm/keystore` dir as default key storage?

File: blockchain/api.go [hunk 2]
```diff
@@ -1,5 +1,290 @@
...
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/sonm-io/core/blockchain/tsc"
+	"github.com/sonm-io/core/blockchain/tsc/api"
+	"github.com/sonm-io/core/blockchain/utils"
+	"math/big"
+)
+
+const ethEndpoint string = "https://rinkeby.infura.io/00iTrs5PIy0uGODwcsrb"
+
+var ethClient *ethclient.Client = nil
```
Comment (sshaman1101) [inline blockchain/api.go]: Store `ethClient` as part of `BlockchainAPI` struct, no reason to keep it as the variable into the package.

File: blockchain/api.go [hunk 3]
```diff
@@ -1,5 +1,290 @@
...
+	WorkTime          *big.Int
+	EndTime           *big.Int
+}
+
+type BlockchainAPI struct {
+	prv    *ecdsa.PrivateKey
+	txOpts *bind.TransactOpts
+}
+
+func NewBlockchainAPI(key *ecdsa.PrivateKey, gasPrice *big.Int) (bch *BlockchainAPI, err error) {
```
Comment (sshaman1101) [inline blockchain/api.go]: Remove named return parmas, please

File: blockchain/api.go [hunk 4]
```diff
@@ -1,5 +1,290 @@
...
+	err = initEthClient()
+	if err != nil {
+		return nil, err
+	}
+
+	txOpts := bind.NewKeyedTransactor(key)
+
+	if gasPrice == nil {
+		gasPrice = big.NewInt(20 * 1000000000)
```
Comment (sshaman1101) [inline blockchain/api.go]: move default values for gas price and gas limit to constants

File: blockchain/api.go [hunk 5]
```diff
@@ -1,5 +1,290 @@
...
+		txOpts: txOpts,
+	}
+	return bch, nil
+}
+
+// ----------------
+// Deals appearance
+// ----------------
+
+func connectDeals() (*api.Deals, error) {
```
Comment (sshaman1101) [inline blockchain/api.go]: connect deals with what? :)

File: blockchain/example/example.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"github.com/sonm-io/core/accounts"
+	"github.com/sonm-io/core/blockchain"
+	"log"
+	"math/big"
```
Comment (sshaman1101) [inline blockchain/example/example.go]: sort imports

Comment (Sokel) [reply blockchain/example/example.go]: connect Deals blockchain api with deployed contract-)

**Pull request merged**